src/archives.c

```diff
@@ -0,0 +1,157 @@
+/*
+ *   Software Updater - client side
+ *
+ *      Copyright © 2012-2017 Intel Corporation.
+ *
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation, version 2 or later of the License.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *   Authors:
+ *         Matthew Johnson <mathew.johnson@intel.com>
+ *
+ */
+
+#include <archive.h>
+#include <archive_entry.h>
+#include <stdlib.h>
+
+#include "swupd.h"
+
+/* archive_check_err(ar, ret)
+ *
+ * Check and print archive errors */
+static int archive_check_err(struct archive *ar, int ret)
+{
+	if (ret < ARCHIVE_OK) {
+		/* print error */
+		fprintf(stderr, "%s\n", archive_error_string(ar));
+	}
+
+	if (ret < ARCHIVE_WARN) {
+		/* error was worse than a warning */
+		return 1;
+	}
+
+	/* just a warning */
+	return 0;
```

Comment:
Good catch, since `ARCHIVE_RETRY` indicates the operation failed, but is -10 while `ARCHIVE_WARN` is -20, so would not be caught by this.

I think we should just treat `ARCHIVE_RETRY` as an extraction failure rather than trying to figure out how many times we want to continue retrying for each separate operation.

Reply:
That sounds fine to me. Implementing extraction retries based on this status code sounds overkill.